moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -0,0 +1,250 @@
+        }
+        return payload
+    }
+
+    override fun identify(payload: IdentifyEvent): BaseEvent {
+        try {
+            super.identify(payload)
+            Logger.print { "$tag identify(): in" }
+            val traits: Traits = payload.traits
+            if (traits.isNotEmpty()) {
+                integrationHelper.trackUserAttribute(
+                    filterOutNonNullTraits(traits),
+                    instanceId
+                )
+                val address = traits[USER_TRAIT_USER_ADDRESS]
+                if (address != null && address.jsonObject.isNotEmpty()) {
+                    val city = address.jsonObject.getString(USER_TRAIT_USER_ADDRESS_CITY)
+                    if (!city.isNullOrEmpty()) {
+                        MoEAnalyticsHelper.setUserAttribute(
+                            application.applicationContext,
+                            USER_TRAIT_USER_ADDRESS_CITY,
+                            city,
+                            instanceId
+                        )
+                    }
+                    val country = address.jsonObject.getString(USER_TRAIT_USER_ADDRESS_COUNTRY)
+                    if (!country.isNullOrEmpty()) {
+                        MoEAnalyticsHelper.setUserAttribute(
+                            application.applicationContext,
+                            USER_TRAIT_USER_ADDRESS_COUNTRY,
+                            country,
+                            instanceId
+                        )
+                    }
+                    val state = address.jsonObject.getString(USER_TRAIT_USER_ADDRESS_STATE)
+                    if (!state.isNullOrEmpty()) {
+                        MoEAnalyticsHelper.setUserAttribute(
+                            application.applicationContext,
+                            USER_TRAIT_USER_ADDRESS_STATE,
+                            state,
+                            instanceId
+                        )
+                    }
+                }
+            }
+            val location = payload.traits[USER_TRAIT_USER_LOCATION]
+            if (location != null && !location.jsonObject.isEmpty()) {
+                MoEAnalyticsHelper.setUserAttribute(
+                    application.applicationContext, USER_ATTRIBUTE_USER_LOCATION, GeoLocation(
+                        location.jsonObject.getDouble(USER_TRAIT_USER_LOCATION_LATITUDE) ?: 0.0,
+                        location.jsonObject.getDouble(USER_TRAIT_USER_LOCATION_LONGITUDE) ?: 0.0
+                    ), instanceId
+                )
+            }
+            Logger.print { "$tag identify(): out" }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag identify(): " }
+        }
+
+        return payload
+    }
+
+    override fun reset() {
+        try {
+            super.reset()
+            Logger.print { "$tag reset(): in" }
+            logoutUser(application.applicationContext, instanceId)
```

Comment:
Don't use static import for functions of the SDKs

Reply:
Done